### PR TITLE
Improve Edge Server template app DX

### PR DESCRIPTION
**Pull request by krollins-mdb**
This PR makes the following changes to the Edge Server template app:
- Added a quick start option. After setting up an app services app and setting up Edge Server, at the project root, users can run `npm run install` to install the Node server and React client dependencies. Then they can run `npm run start` to start the Edge Server, Node server, and React client. `npm run shut-down` shuts down all three pieces.
- Changed the Express server port from 5000 to 5055. Also updated the React client to refer to 5055. This is to avoid AirPlay on MacOS, which sometimes takes over port 5000.
- Hard-code an owner_id of {{"edge"}} for the Express client.

Notable not doing:
- Abstract port value so that the Express server and React client share the same value. This became a harder problem to solve than expected. If users need to change the port value, they need to change it in the Express server and React client code. This is already documented.

### CODE REVIEW

File: other/edge-server/README.md
Comment (dacharyc) [inline other/edge-server/README.md:310]:
```suggestion
This example app uses port `5055` by default.
```

File: other/edge-server/README.md
Comment (dacharyc) [inline other/edge-server/README.md:268]: Oof. I am realizing now this is even more of a PITA since you've made it generally nicer about not having to install dependencies and start the servers individually.

At the very least, since this README is now above the `client` directory and the `node-server` directory isn't visible, we should probably add that detail.

But it would really be nicer to have some higher-level thing people can config instead of having to dig deeper to configure stuff.
```suggestion
Add a `.env` file inside the `client/node-server` directory with the details
```

(We should probably also remove the lines above about cding into `node-server` and installing dependencies, since you've now got that higher-level script that manages installing all the dependencies.)

File: other/edge-server/README.md
Comment (dacharyc) [inline other/edge-server/README.md:279]: We can probably remove these lines about running the Express Server since that is done now at the higher level.

Comment (dacharyc) [reply other/edge-server/README.md:279]: Can't comment below, but we should update the path in ln 269/292 to show that `node-server` is in the `client/` directory, and in ln 273/296 to say instead "at the root of your `client/node-server` directory.

File: other/edge-server/README.md
Comment (dacharyc) [inline other/edge-server/README.md:313]:
```suggestion
`client/node-server` directory. If you change this port, you must also change
```

File: other/edge-server/README.md
Comment (dacharyc) [inline other/edge-server/README.md:315]:
```suggestion
Change the port in `client/react-client/src/endpoints.ts` on line 4 to match
```

File: other/edge-server/README.md
Comment (dacharyc) [inline other/edge-server/README.md:338]: We can probably just delete this section about installing dependencies and starting the React server, since that can now be managed at the higher level with your new package.json.

(Also, we should document that as the way to install the dependencies and run the servers.)

File: other/edge-server/README.md
Comment (dacharyc) [inline other/edge-server/README.md:435]: I would move this section up and remove the stuff about installing dependencies for the node server and the React client separately. This is the happy path and we want to present it front-and-center.

File: other/edge-server/README.md
Comment (dacharyc) [inline other/edge-server/README.md:401]: We've added back the `owner_id` here, but it's not reflected in the schema below. We should update this schema to match the one we're now using in the backend.

File: other/edge-server/README.md
Comment (dacharyc) [inline other/edge-server/README.md:58]: I _think_ we can take this note out now. By adding the `owner_id` back, we should have made it interoperable with the Device SDK template apps. (We should also test this backend with the Device SDK template apps.)

You must still use `edge-server.todo` to _create_ a template app that the Edge Server template can connect to, but this should now work with the Device SDK templates.

**Review by dacharyc — COMMENTED** [review]
Back to you with some comments re: the README.

Also, I'm curious - have you tested this with a Device SDK template app? It should now be interoperable, but we'll want to confirm that before we merge this.

**Review by dacharyc — COMMENTED** [review]
Ah! Just realized our workflow also only pushes files that are _in_ the `client` directory to the artifact repo: https://github.com/mongodb-university/realm-template-apps/blob/main/.github/workflows/push-artifact-edge-server.yml

And the artifact repo looks like this: https://github.com/mongodb/template-app-edge-server

So if you want to have the README.md, the package-lock.json and the package.json all go up to the artifact repo, you'll need to edit the workflow. And I assume you'll also want to exclude the `backend` directory since we don't necessarily want that in the artifact repo.

This may be a good case for creating a `generated` directory and using Bluehawk copy to only copy over the files we want, excluding the `backend`, and push _that_ up to the artifact repo. I'll let you decide how you want to handle that.

File: other/edge-server/README.md
Comment (krollins-mdb) [reply other/edge-server/README.md:268]: Yeah, would be nice to streamline this more. I'll do some more thinking about a good approach in the future.

File: other/edge-server/README.md
Comment (krollins-mdb) [reply other/edge-server/README.md:338]: I reframed this section and the Express section as alternatives to the quick start and I link to the quick start above. The info in these sections still works if people want to do it on their own.

Comment (krollins-mdb) [reply other/edge-server/README.md:338]: Just kidding. I've merged the Express and React sections and made it more of the happy path.

**Comment by krollins-mdb** [comment]
Excellent point, Dachary! I don't know why I had it in my head that I needed this stuff at the `edge-server` level. I moved everything to the `client` directory and tested again. Should be good now.

**Review by dacharyc — APPROVED** [review]
Looking good! Just a couple of other small suggestions to clarify some of the backend stuff, but that's not really a blocker.

**Pull request merged**
